### ISSUE TITLE
Proposal for changes in CLI

### ISSUE THREAD

**Issue by pksunkara** [apiaryio/drafter#158]
I propose a few changes to the existing CLI of drafter executable compiled from the master branch. The current structure of the CLI is:

```
usage: drafter [options] ... <input file>

API Blueprint Parser
If called without <input file>, 'drafter' will listen on stdin.

options:
  -o, --output               save output AST into file (string [=])
  -f, --format               output format of the AST (yaml|json) (string [=yaml])
  -t, --type                 type of the AST (refract|ast) (string [=refract])
  -s, --sourcemap            export sourcemap AST into file (string [=])
  -r, --refract-sourcemap    produce sourcemap for refract AST
  -h, --help                 display this help message
  -v, --version              print Drafter version
  -l, --validate             validate input only, do not print AST
  -u, --use-line-num         use line and row number instead of character index when printing annotation
```

There is no way to get the whole parse result from this. The CLI only produces Normal/Refract AST, and if prodded further produces source maps.

Since we change the behaviour of the CLI anyway, I propose to change the following things too.

1. By default, CLI should give the whole parse result for both Normal and Refract AST.
2. Merge both `-r` and `-s <file>` into a single `-s` option which when exists produces source maps in the parse result.

The following suggestion is not related to the above ones, but might alleviate some confusion.

1. Change `ast` value for `-t` option to `normal`. Drafter calls them Normal AST and Refract AST. So, the `ast` value might be confusing for some people.

 ```diff
+  -t, --type                 type of the AST (refract|ast) (string [=refract])
-  -t, --type                 type of the AST (refract|normal) (string [=refract])
```

**Comment by pksunkara** [comment]
@apiaryio/api-blueprint Let's discuss and finalize this first before releasing drafter v2.

**Comment by klokane** [comment]
In general it is backward compatibility break. So there is question, how many user use CLI?
From another point of view, there are already changes in serialization format (refract).
I have no strong opinion about it.
What does mean @zdne?

**Comment by zdne** [comment]
@pksunkara so the reason for this is mainly to get the whole parse result from the CLI tool?

I am OK with merging `-r` and `-s`. 

I am somewhat OK with providing the whole parser result but I miss **the reasoning** for doing so. 

I would refrain from changing  `-t` value `ast` value to `normal`. There is nothing "normal" about the AST 😀 please keep in mind that we do not want to deprecate it in the future.

**Comment by zdne** [comment]
> Let's discuss and finalize this first before releasing drafter v2.

Regardless of this issue I would like to discuss with you and @kylef why do you think using the milestones like this is a good idea.

**Comment by pksunkara** [comment]
> Regardless of this issue I would like to discuss with you and @kylef why do you think using the artificial milestones like this is a good idea.

I consider this a breaking change and since we are releasing a major version anyway, I think it's best to incorporate this change before v2 is released so that we can try not to break anything after v2 is released.

@zdne The reason behind the proposal is the following.

> There is no way to get the whole parse result from this. The CLI only produces Normal/Refract AST, and if prodded further produces source maps.

Given a parse result, people can get AST and source maps, but not the reverse.

The current CLI structure feels like hacks placed upon hacks. First there was AST, then we added source maps using a separate argument, then we added refract source maps using another argument.

I feel like it is time we clean the mess up a bit.
